### ISSUE TITLE
loss=nan when training YOLO3.0

### ISSUE THREAD

**Issue by xuezu29** [dmlc/gluon-cv#278]
When I'm training YOLO3.0 by my data set, the loss gets 'nan'. But using the same data set for Gluoncv's faster-rcnn and ssd it's all OK. it's so puzzled. I just changed the anchor size and image size, others use default values.

**Comment by zhreshold** [comment]
what if you don't change anchor size?

**Comment by xuezu29** [comment]
@zhreshold I've tried it ,still 'nan'. When I change the image size I keep the aspect ratio of the image. it is the key?  When I used SSD also keeping the aspect ratio of the image, but would  modify the validate code, the same goes for the yolo3.

**Comment by xuezu29** [comment]
I guess if any layer of the three predicted stage have no objects the nan would be appear? 
![001](https://user-images.githubusercontent.com/10649774/44708714-67734480-aada-11e8-827c-f41835779de9.png)

**Comment by zhreshold** [comment]
1. If you are using original aspect ratio, what is the batch size then? 
2. The total number of normalization factor is calculated across batch size, number of total predictions.

There's always going to be a best match, but I don't quick understand what you have modified in this example particularly.

**Comment by xuezu29** [comment]
@zhreshold Hi, thank you for your help. I noticed that  the value of warmup_epochs led this problem in my case. But I still have a question, When I decrease the warmup_epochs would be  ok, why not increase?  In theory ,big warmup_epochs means that low "lr" would be keep more epoch, shouldn't be better?

**Comment by nicklhy** [comment]
Got the same problem when running `train_yolo3.py` on VOC ! (mxnet 1.3.0.post0 and the newest gluoncv version)
In my case, I found the nan disappeared if I remove `net.hybridize()` in both `train()` and `validate()`. Any ideas about this problem?

**Comment by fadedfat3** [comment]
@nicklhy After i removed `net.hybridize()`, the nan disappeared too. Maybe it was because that hybridize() convert NDArray to Symbol.

**Comment by zhreshold** [comment]
Just an update, the root cause is found and fix has been merged to master: https://github.com/apache/incubator-mxnet/pull/14209

By using master/nightly built pip package hopefully you won't meet same problem any more

**Comment by rohit167** [comment]
I am still facing this issue and receiving NAN's while training yolo3 darknet53 on custom data set with 608x608 fixed resolution.

batch size: 32
warmup-epochs: 2
Mxnet version: 1.6.0'

Any leads to resolve this?

**Comment by zhreshold** [comment]
> I am still facing this issue and receiving NAN's while training yolo3 darknet53 on custom data set with 608x608 fixed resolution.
> 
> batch size: 32
> warmup-epochs: 2
> Mxnet version: 1.6.0'
> 
> Any leads to resolve this?

Try warm-up learning rate for the first 2 epochs

**Comment by rohit167** [comment]
Lowering the learning rate did the trick for me. 
Thanks @zhreshold I will try your suggestions with earlier settings as well.
